framework/src/nodalkernels/ADNodalKernel.C

```diff
@@ -25,10 +25,10 @@ ADNodalKernel::validParams()
 ADNodalKernel::ADNodalKernel(const InputParameters & parameters)
   : NodalKernelBase(parameters), _u(_var.adDofValues())
 {
-  if (isParamValid("save_in"))
+  if (isParamSetByUser("save_in"))
```

Comment:
I don't quite trust isParamSetByUser. What happens if the param is set by an action?

Reply:
As above, why not keep this file unchanged? The semantics here are still the same. If no default is provided and the user didn't set the parameter we should use `isParamValid`. This code is not relying on an implict default, so let's not change it.

Reply:
Is there a better way to do it along with providing default value? Check size?